covalent_dispatcher/_service/app.py

```diff
@@ -183,17 +104,29 @@ async def cancel(request: Request) -> str:
         Fast API Response object confirming that the dispatch
         has been cancelled.
     """
-    data = await request.body()
-    dispatch_id = data.decode("utf-8")
+    try:
+        data = await request.body()
+        dispatch_id = data.decode("utf-8")
 
-    dispatcher.cancel_running_dispatch(dispatch_id)
-    return f"Dispatch {dispatch_id} cancelled."
+        dispatcher.cancel_running_dispatch(dispatch_id)
+        return f"Dispatch {dispatch_id} cancelled."
+    except Exception as e:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Failed to cancel workflow: {e}",
+        ) from e
 
 
 @router.get("/db-path")
 def db_path() -> str:
-    db_path = DispatchDB()._dbpath
-    return json.dumps(db_path)
+    try:
+        db_path = DispatchDB()._dbpath
+        return json.dumps(db_path)

```

Comment:
@FyzHsn while we're here let's remove this route,  the `_db_path` utility in results manager utilts, and the corresponding `test_db_path` test